python/tank/platform/bundle.py

```diff
@@ -940,7 +959,21 @@ def __resolve_hook_path(self, settings_name, hook_expression):
             hooks_folder = os.path.join(engine.disk_location, "hooks")
             path = hook_expression.replace("{engine}", hooks_folder)
             path = path.replace("/", os.path.sep)
-        
+
+        elif hook_expression.startswith("{show}"):
+            # this is expected to be used only when there are show overrides, and hence
+            # possible declaration of correct hook name in the sgtk_config preferences files.
+            path = hook_expression.replace("{show}/", "")
+            # check the show tree for the hook!
+            resolved_hook_path_list = self.expand_show_tree(path,
+                                                            subdirectories=["SHARED/etc/sgtk/hooks", "etc/sgtk/hooks"])
+            if resolved_hook_path_list:
+                # we found a relavant list of hooks on the show!
+                # list is in the order Test_User:Shot:Seq:Project
+                path = resolved_hook_path_list[0]
```

Comment:
Hmm...this is an interesting use case. Are you sure you wouldn't want this to return all found paths? So that shot inherits from sequence inherits from project?

Reply:
Makes sense to me, I can leave the list intact. DCC versions of multi-publish can leverage it, since the bundle context and item context will be same. This won't be usable by ingest though since bundle and item contexts are usually different.

Reply:
Pretty sure this should work in that case as well, since the Item's settings are resolved from a temporarily instantiated bundle using the Item's context. But let me know if it doesn't...